### PR TITLE
Migrate assertEquals with array parameter to assertArrayEquals

### PR DESCRIPTION
**Pull request by puskarpeter**
## What's changed?

Added a test case to cover the mentioned use case.

## What's your motivation?

- assertEquals in JUnit4 accepts Object[], to preserve the behaviour we need to rewrite it into assertArrayEquals
https://junit.org/junit4/javadoc/4.13/org/junit/Assert.html#assertEquals(java.lang.Object%5B%5D,%20java.lang.Object%5B%5D)
### Checklist
- [ ] I've added unit tests to cover both positive and negative cases
- [x] I've added the license header to any new files through `./gradlew licenseFormat`
- [x] I've used the IntelliJ auto-formatter on affected files
- [x] I've updated the documentation (if applicable)

### CODE REVIEW

File: src/test/java/org/openrewrite/java/testing/junit5/AssertToAssertionsTest.java [hunk 1]
```diff
@@ -462,4 +462,35 @@ void test() {
           )
         );
     }
+
+    // edge case for deprecated use of assertEquals
+    // https://junit.org/junit4/javadoc/4.13/org/junit/Assert.html#assertEquals(java.lang.Object%5B%5D,%20java.lang.Object%5B%5D)
+    @Issue("#383")
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/testing/junit5/AssertToAssertionsTest.java:468]:
```suggestion
    @Issue("https://github.com/openrewrite/rewrite-testing-frameworks/pull/384")
```

File: src/test/java/org/openrewrite/java/testing/junit5/AssertToAssertionsTest.java [hunk 2]
```diff
@@ -462,4 +462,35 @@ void test() {
...
+    @Test
+    void assertEqualsWithArrayArgumentToAssertArrayEquals() {
+        //language=java
+        rewriteRun(
+          spec -> spec.typeValidationOptions(TypeValidation.none()),
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/testing/junit5/AssertToAssertionsTest.java:473]:
```suggestion
```
Ideally this is not necessary; in rare cases where we want to make changes despite missing types we add that to some tests, but it could potentially break recipe chaining, as the next recipe will then not have the correct types available.

Comment (puskarpeter) [reply src/test/java/org/openrewrite/java/testing/junit5/AssertToAssertionsTest.java]: Oh okay, I just copy-pasted it from the previous example without knowing what it really does exactly. Thanks for the explanation.

Comment (timtebeek) [reply src/test/java/org/openrewrite/java/testing/junit5/AssertToAssertionsTest.java]: You're welcome; thought as much seeing as the one just above had the same; figured point it out with a bit of background. :)

**Review by timtebeek — APPROVED** [review]
